src/framework/components/anim/property-locator.js

```diff
@@ -0,0 +1,50 @@
+Object.assign(pc, function () {
+    /**
+     * @class
+     * @name pc.AnimPropertyLocator
+     * @classdesc The AnimProperyLocator encodes and decodes paths to properties in the scene hierarchy.
+     * @description Create a new AnimPropertyLocator.
+     */
+    var AnimPropertyLocator = function () {
+    };
+    Object.assign(AnimPropertyLocator.prototype, {
+        /**
+         * @function
+         * @name pc.AnimPropertyLocator#encode
+         * @description Converts a locator array into its string version
+         * @param {Array} locator - The property location in the scene defined as an array
+         * @returns {string} The locator encoded as a string
+         * @example
+         * // returns 'spotLight/light/color.r'
+         * encode([['spotLight'], 'light', ['color', 'r']]);
+         */
+        encode: function (locator) {
+            return pc.AnimBinder.joinPath([
+                pc.AnimBinder.joinPath(locator[0]),
+                locator[1],
+                pc.AnimBinder.joinPath(locator[2])
+            ], '/');
```

Comment:
you can't use different splitter characters for the inner and outer levels (the result won't be correctly escaped), but actually there is no need. just use the same splitter at both levels (and I'd say remove option to customize the splitter character from the AnimBinder functions).

Reply:
actually the two levels will be correctly escaped. so even though it's unnecessary to have different splitters it does work fine and I'm happy leaving it like this.